src/Microsoft.Diagnostics.Monitoring.StartupHook/Exceptions/Pipeline/ExceptionPipeline.cs

```diff
@@ -13,6 +14,8 @@ internal sealed class ExceptionPipeline :
 
         private long _disposedState;
 
+        private int _processingState;
```

Comment:
Could Semaphore/SemaphoreSlim be used instead?

Reply:
I don't think I can use these easily since I don't really have a fixed count here. I ultimately landed on an Interlocked reader/writer lock solution.